lib/auth/access_test.go

```diff
@@ -173,6 +178,774 @@ func TestUpsertDeleteDependentRoles(t *testing.T) {
+			wantErr: func(t require.TestingT, err error, i ...interface{}) {
+				require.True(t, trace.IsAccessDenied(err))
+			},
+		},
+		{
+			name: "update role where clause",
+			userRole: newRole(t, "urole", nil, rcWithRoleRuleWhere([]string{types.VerbUpdate},
+				builder.Equals(
+					builder.Identifier(`resource.metadata.labels["label"]`),
+					builder.String("value"),
+				).String(),
+			), types.RoleConditions{}),
+			startingRole: newRole(t, "update", map[string]string{
+				"label": "value",
+			}, types.RoleConditions{}, types.RoleConditions{}),
+			updateRole: newRole(t, "update", map[string]string{
+				"label": "value",
+			}, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			wantErr: require.NoError,
+			want: newRole(t, "update", map[string]string{
+				"label": "value",
+			}, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+		},
+		{
+			name: "update role where clause denied",
+			userRole: newRole(t, "urole", nil, rcWithRoleRuleWhere([]string{types.VerbUpdate},
+				builder.Equals(
+					builder.Identifier(`resource.metadata.labels["label"]`),
+					builder.String("value"),
+				).String(),
+			), types.RoleConditions{}),
+			startingRole: newRole(t, "update", nil, types.RoleConditions{}, types.RoleConditions{}),
+			updateRole:   newRole(t, "update", nil, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			wantErr: func(t require.TestingT, err error, i ...interface{}) {
+				require.True(t, trace.IsAccessDenied(err))
+			},
+		},
+		{
+			name: "update role old role doesn't match where",
+			userRole: newRole(t, "urole", nil, rcWithRoleRuleWhere([]string{types.VerbUpdate},
+				builder.Equals(
+					builder.Identifier(`resource.metadata.labels["label"]`),
+					builder.String("value"),
+				).String(),
+			), types.RoleConditions{}),
+			startingRole: newRole(t, "update", nil, types.RoleConditions{}, types.RoleConditions{}),
+			updateRole: newRole(t, "update", map[string]string{
+				"label": "value",
+			}, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			wantErr: func(t require.TestingT, err error, i ...interface{}) {
+				require.True(t, trace.IsAccessDenied(err))
+			},
+		},
+		{
+			name: "update role new role doesn't match where",
+			userRole: newRole(t, "urole", nil, rcWithRoleRuleWhere([]string{types.VerbUpdate},
+				builder.Equals(
+					builder.Identifier(`resource.metadata.labels["label"]`),
+					builder.String("value"),
+				).String(),
+			), types.RoleConditions{}),
+			startingRole: newRole(t, "update", map[string]string{
+				"label": "value",
+			}, types.RoleConditions{}, types.RoleConditions{}),
+			updateRole: newRole(t, "update", nil, rcWithRoleRule(services.RW()...), types.RoleConditions{}),
+			wantErr: func(t require.TestingT, err error, i ...interface{}) {
+				require.True(t, trace.IsAccessDenied(err))
+			},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
```

Comment:
same comment re: parallel subtests, right now they're serial and the test := test is not necessary (aren't we all looking forward to go 1.22 eliminating this forever)